packages/account-api/src/api/multichain/account.ts

```diff
@@ -26,7 +24,7 @@ export type MultichainAccountId = `${MultichainAccountWalletId}/${number}`; // U
 /**
  * A multichain account that holds multiple accounts.
  */
-export class MultichainAccount<Account extends KeyringAccount>
+export class MultichainAccount<Account extends Bip44Account<KeyringAccount>>
```

Comment:
Enforcing `Bip44Account` here, allows us to use the new typed options pretty easily (the `options.entropy.groupIndex` is guaranteed to be defined with this)

Also, this forces the `AccountProvider` to only provide `Bip44Account` which should also prevent unwanted accounts during filtering.